src/algebra/polynomial/big_operators.lean

```diff
@@ -58,39 +60,60 @@ begin
 end
 
 /-- The degree of a product of polynomials is equal to the product of the degrees, provided that the product of leading coefficients is nonzero.
-See `nat_degree_prod_eq` (without the `'`) for a version for integral domains, where this condition is automatically satisfied. -/
-lemma nat_degree_prod_eq' (h : ∏ i in s, (f i).leading_coeff ≠ 0) :
+See `nat_degree_prod` (without the `'`) for a version for integral domains, where this condition is automatically satisfied. -/
+lemma nat_degree_prod' (h : ∏ i in s, (f i).leading_coeff ≠ 0) :
   (∏ i in s, f i).nat_degree = ∑ i in s, (f i).nat_degree :=
 begin
   classical,
   revert h, induction s using finset.induction with a s ha hs, { simp },
   rw [prod_insert ha, prod_insert ha, sum_insert ha],
-  intro h, rw polynomial.nat_degree_mul_eq', rw hs,
+  intro h, rw polynomial.nat_degree_mul', rw hs,
   apply right_ne_zero_of_mul h,
   rwa polynomial.leading_coeff_prod', apply right_ne_zero_of_mul h,
 end
 
-lemma monic_prod_of_monic :
-  (∀ a : α, a ∈ s → monic (f a)) → monic (∏ i in s, f i) :=
-by { apply prod_induction, apply monic_mul, apply monic_one }
-
-lemma nat_degree_prod_eq_of_monic [nontrivial R] (h : ∀ i : α, i ∈ s → (f i).monic) :
+lemma nat_degree_prod_of_monic [nontrivial R] (h : ∀ i ∈ s, (f i).monic) :
   (∏ i in s, f i).nat_degree = ∑ i in s, (f i).nat_degree :=
 begin
-  apply nat_degree_prod_eq',
-  suffices : ∏ (i : α) in s, (f i).leading_coeff = 1, { rw this, simp },
+  apply nat_degree_prod',
+  suffices : ∏ i in s, (f i).leading_coeff = 1, { rw this, simp },
   rw prod_eq_one, intros, apply h, assumption,
 end
 
 end comm_semiring
 
+section comm_ring
+variables [comm_ring R]
+
+open monic
+-- Eventually this can be generalized with Vieta's formulas
+-- plus the connection between roots and factorization.
+lemma next_coeff_prod_X_sub_C [nontrivial R] {s : finset ι} (f : ι → R) :
+next_coeff ∏ i in s, (X - C (f i)) = -∑ i in s, f i :=
```

Comment:
Maybe easier to find as `prod_X_sub_C_next_coeff`? Can you move the `next_coeff` to use projection notation?